### PR TITLE
added multi_dot to jax numpy linalg and tested

### LINKED ISSUE #16620: multi_dot

### PR DESCRIPTION
**Pull request by Principor**
Close #16620

### CODE REVIEW

**Comment by bipinKrishnan** [comment]
Hey @Principor , I think it's best to give a generic name to `_get_dtype_inputs_for_multi_dot` function and move it to `array_helpers.py` :slightly_smiling_face: . Also, is the input array to the function constrained to just floating type?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py [hunk 1]
```diff
@@ -834,3 +834,52 @@ def test_jax_numpy_cond(
         x=x[0],
         p=p,
     )
+
+
+@st.composite
+def _get_dtype_inputs_for_multi_dot(draw):
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py:840]: I think it's best to give a generic name to this function and move it to `array_helpers.py`.

**Comment by bipinKrishnan** [comment]
Hey @Principor, can you merge the master and push again. Seems like some of the tests not related to your changes are failing currently :slightly_smiling_face:

File: ivy/functional/frontends/jax/numpy/linalg.py [hunk 2]
```diff
@@ -152,3 +154,12 @@ def cond(x, p=None):
...
     r = ivy.where(orig_nan_check, ivy.where(nan_mask, ivy.inf, r), r)
     return r
+
+
+@to_ivy_arrays_and_back
+def multi_dot(arrays, /, *, precision=None):
+    dtype = arrays[0].dtype
+    for i in range(1, len(arrays)):
+        dtype = jax.numpy.promote_types(dtype, arrays[i].dtype)
+    arrays = list(array.astype(dtype) for array in arrays)
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/jax/numpy/linalg.py:164]: We can avoid this explicit dtype casting since the other ivy frontends doesn't employ this.

Comment (Principor) [reply ivy/functional/frontends/jax/numpy/linalg.py:164]: Sorry, I'm not sure exactly what you mean here. Without the type casting some of the tests where failing because the output had the wrong dtypes. Should I just admit the errors and accept the test failures?

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 3]
```diff
@@ -1047,7 +1047,6 @@ def array_indices_put_along_axis(
 
     Parameters
     ----------
-
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:1050]: I think these are not related to the function you are implementing, so I you can revert it back :slightly_smiling_face:

File: ivy/functional/frontends/jax/numpy/linalg.py (hunk 2 above)
Comment (bipinKrishnan) [reply ivy/functional/frontends/jax/numpy/linalg.py:138]: In that case, don't you think it should be present with other frontends as well? I think we should fix the issues that occurs when not type casting(if anything comes in). Let's see how it goes :slightly_smiling_face:

Comment (bipinKrishnan) [reply ivy/functional/frontends/jax/numpy/linalg.py:138]: Also it's not allowed to import jax to the frontend I guess. May be this caused other tests not related to your pr to fail.

Comment (bipinKrishnan) [reply ivy/functional/frontends/jax/numpy/linalg.py:138]: I would suggest you try locally and if errors still persists, you could use type casting but instead of jax, use some ivy function to do this.

Comment (Principor) [reply ivy/functional/frontends/jax/numpy/linalg.py:138]: That does make sense, sorry I should have realised that earlier. Do you know if there are there any ivy functions that emulate a frontends' dtype promotion rules? I can't seem to find any

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by bipinKrishnan** [comment]
ivy-gardener

**Comment by bipinKrishnan** [comment]
Hey @Principor , I would suggest you install pre-commit and lint the code before committing :slightly_smiling_face:

File: ivy/functional/frontends/jax/numpy/linalg.py (hunk 2 above)
Comment (bipinKrishnan) [reply ivy/functional/frontends/jax/numpy/linalg.py:138]: I think there is something like `promote_type`. You could search the docs for something like this.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
